ui/src/views/actions/thunks.ts

```diff
@@ -112,8 +115,50 @@ export const getViewForTimeMachine = (
         view,
       })
     )
+  } catch (error) {
+    console.error(error)
+    dispatch(notify(copy.getViewFailed(error.message)))
+    dispatch(setView(cellID, RemoteDataState.Error))
+  }
+}
+
+export const setQueryResultsByQueryID = (queryID: string) => (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const state = getState()
+    const files = state.queryCache.queryResultsByQueryID[queryID]
+    if (files) {
+      dispatch(setQueryResults(RemoteDataState.Done, files, null, null))
+      return
+    }
     dispatch(executeQueries())
   } catch (error) {
+    // if the files don't exist in the cache, we want to execute the query
+    dispatch(executeQueries())
+  }
+}
+
+export const setQueryResultsForCell = (
+  dashboardID: string,
+  cellID: string,
+  timeMachineID: TimeMachineID
+) => (dispatch, getState: GetState): Promise<void> => {
+  try {
+    dispatch(getViewForTimeMachine(dashboardID, cellID, timeMachineID))
```

Comment:
I'm worried about race conditions here. We can not await the results of a thunk, we just dispatch it, and so if we are depending on its results as we are... on line 151, we're hoping that getViewForTimeMachine has done its thing already and we can get the view. The solutions is to integrate the `getViewForTimeMachine` and `setQueryResultsForCell` functions in to one.

Reply:
I hear what you're saying, it's definitely a valid concern, but it seems like it might be overstepping the intent of the function. I'm curious as to how moving the dispatches within the scope of `setQueryResultsForCell` would prevent a race condition?